# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/bytebymanas/fly-rank-ML/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


**Lane 2 – Refresh / Content Opportunity Scoring**

### Why I chose this lane

I selected the **Refresh / Content Opportunity Scoring** lane because organizations often manage thousands of content pages but have limited time and resources to review them manually. Instead of treating every page equally, I want to build a system that identifies which pages deserve attention first based on observable search and engagement signals.

The goal is not to automatically update content, but to support better editorial decision-making by producing a ranked review queue. This combines business value with machine learning by helping teams focus their efforts on pages that have the highest potential for improvement.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Decision

This project improves the decision of **which content pages should be reviewed first**. Since content teams have limited time and resources, reviewing every page is not practical. The goal is to prioritize pages that are most likely to benefit from a content refresh based on observable search and engagement signals.

### Who Acts on It?

The primary users are **content managers, SEO specialists, and editorial teams**. They use the ranked recommendations to decide where to focus their review and optimization efforts.

### Action

Based on the recommendations, the team can:

- Refresh outdated content
- Improve page quality or structure
- Rewrite titles and meta descriptions
- Expand thin content
- Monitor pages that do not yet require immediate action

### Cost of a Wrong Recommendation

A **false positive** recommends a healthy page for review, causing editors to spend time and resources on content that does not require immediate attention.

A **false negative** fails to identify a page that is genuinely declining, allowing it to continue losing search visibility and potential traffic.

The objective of this project is to improve the prioritization of limited editorial resources rather than guarantee that refreshing a page will increase its performance.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("/content_refresh_anonymized.csv")

# Basic information
print(f"Total pages: {len(df):,}")

print("\nTrend Distribution:")
print(df["trend_direction"].value_counts())

print("\nImpressions (90 Days):")
print(df["impressions_90d"].describe())

print("\nSessions (90 Days):")
print(df["sessions_90d"].describe())

print("\nAverage CTR:")
print(df["ctr"].mean())

Total pages: 30,000

Trend Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Impressions (90 Days):
count     30000.000000
mean       5200.366300
std       16838.019547
min           1.000000
25%          81.000000
50%         731.000000
75%        3615.250000
max      517715.000000
Name: impressions_90d, dtype: float64

Sessions (90 Days):
count    30000.000000
mean        37.066633
std        107.069131
min          1.000000
25%          2.000000
50%          7.000000
75%         27.000000
max       4345.000000
Name: sessions_90d, dtype: float64

Average CTR:
0.5107333333333334



The starter dataset contains **30,000 content pages**, providing a sufficiently large sample for building and evaluating a content opportunity scoring system.

Out of these, **16,262 pages (approximately 54%) are labeled as having a downward trend**, indicating that a significant portion of the content inventory may require review. This reinforces the need for a system that can prioritize which pages deserve attention first.

The **median page receives 731 impressions over the last 90 days**, while the highest-performing page receives **517,715 impressions**, showing that search visibility varies greatly across pages. Since not every page contributes equally to search performance, a ranked review queue is more practical than reviewing every page manually.

These observations suggest that the dataset contains meaningful variation in content performance, making it well suited for building a machine learning system that prioritizes pages for editorial review.

## 4. Careful words: what I can and can't claim


This project is designed as a **decision-support system**. It uses observable search and engagement signals to identify and rank content pages that appear to be good candidates for human review.

The recommendations are based on patterns observed in the available data and are intended to help content managers prioritize limited editorial resources more effectively.

### What this project cannot claim

This project **cannot prove** that refreshing a page will increase traffic or improve rankings. It also cannot predict Google's search algorithm or explain why Google ranks pages in a particular way.

The results should be interpreted as evidence-based recommendations rather than guarantees. Any improvement after a content update would require additional experimentation or causal analysis to establish a cause-and-effect relationship.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.